setup/storage/s3.md

```diff
@@ -5,4 +5,39 @@ sidebar:
   nav: setup
 ---
 
-Documentation for S3 setup with Halyard is coming soon!
+{% include toc %}
+
+Using [S3](https://aws.amazon.com/s3/) as a 
+storage source means that Spinnaker will store all of its persistent data in a
+[Bucket](https://aws.amazon.com/s3/details/). 
+
+## Prerequisites
+
+You need an [AWS Account](https://aws.amazon.com/account/), and an associated
+access key (secret access key & access key id). 
+
+## Editing your Storage Settings
+
+Halyard will create a bucket for you if the bucket you specify doesn't exist
+yet, or if you don't specify one at all. However, if this is the case, you need
+to supply the bucket's region (e.g. `REGION=us-west-1`).
+
+To configure the S3 settings, run the following (notice, do not supply the
+value of `--secret-access-key` on the command line, you will be prompted to
+enter the value on STDIN once the command has started running):
+
+```bash
+hal config storage s3 edit --secret-access-key \
+    --secret-key-id $YOUR_SECRET_KEY_ID \
```

Comment:
Doesn't the key ID generally come first? I'd keep the standard userID -> password ordering.